lib/jekyll-redirect-from/generator.rb

```diff
@@ -22,11 +25,17 @@ def generate(site)
         generate_redirect_to(doc)
       end
 
-      generate_redirects_json
+      if generate_redirects_json?
+        generate_redirects_json 
+      end
     end
 
     private
 
+    def option(key)
+      site.config[CONFIG_KEY] && site.config[CONFIG_KEY][key]
```

Comment:
Prefer [`Hash#dig`](https://ruby-doc.org/core-2.3.0/Hash.html#method-i-dig) from Ruby 2.3 here:
```suggestion
      site.config.dig("redirect_from", key)
```